src/liballoc/pin.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright 2012-2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Types which pin data to its location in memory
+//!
+//! see the [standard library module] for more information
+//!
+//! [standard library module]: ../../std/pin/index.html
```

Comment:
Alternatively, this could re-export the stuff from `core`, so that the standard library module is just a full re-export of the one in liballoc. Isn't that what we usually do?